SteakBot.Core/Modules/MemeModule.cs

```diff
@@ -42,10 +42,60 @@ public async Task AddMeme(string name, string value, string description)
             }
             else
             {
-                replyMessage = "Saving failed! ;(";
+                replyMessage = "Add failed! ;(";
             }
 
             await ReplyAsync(replyMessage, embed: embed?.Build());
         }
+
+        [Command("editMeme")]
+        public async Task EditMeme(string name, string newName, string newValue, string newDescription)
+        {
+            var replyMessage = string.Empty;
+            EmbedBuilder embed = null;
+
+            var type = _memeService.GetMemeType(newValue);
+
+            var result = _memeService.EditCommand(name, new MemeCommand(type, newName, newValue, newDescription));
+
+            if (result)
+            {
+                if (type == MemeResultType.Image)
+                {
+                    embed = new EmbedBuilder
+                    {
+                        ImageUrl = newValue
+                    };
+                }
+                else
+                {
+                    replyMessage = newValue;
+                }
+            }
+            else
+            {
+                replyMessage = "Edit failed! ;(";
+            }
+
+            await ReplyAsync(replyMessage, embed: embed?.Build());
```

Comment:
This is a straight copy of `AddMeme()`. Please pull as much as possible into a reusable method.

Reply:
Corrected.